timeless/sms.py

```diff
@@ -13,6 +15,28 @@ def send(self):
         raise NotImplementedError
 
 
+class RetrySendSMS:
+    """Decorator class for resending requests"""
+    def __init__(self, retry_count=5, timeout=1):
+        self.retry_count = retry_count
+        self.timeout = timeout
+        self.counter = 0
+
+    def __call__(self, fn):
+        @functools.wraps(fn)
+        def decorated(*args, **kwargs):
+            response = fn(*args, **kwargs)
+            response_status_code = response.status_code
+            while (response_status_code > 401 and
```

Comment:
@vladarefiev I think we should only retry in case of 404 and 500. For instance, ``403 FORBIDDEN`` or ``412 PRECONDITION FAILED`` are clearly cases that won't work no matter how many times we retry, since we are doing something wrong.

Reply:
@amihaiemil ok, make sense